internal/server/data/data.go

```diff
@@ -390,3 +405,35 @@ func InfraConnectorIdentity(db ReadTxn) *models.Identity {
 	}
 	return connector
 }
+
+const slowQueryThreshold = time.Second
+
+func logQuery(query string, err error, startedAt time.Time, rows int64) {
+	level := zerolog.TraceLevel
+
+	elapsed := time.Since(startedAt)
+	switch {
+	case errors.Is(err, sql.ErrNoRows), errors.Is(err, gorm.ErrRecordNotFound):
+		level = zerolog.WarnLevel
+	case elapsed > slowQueryThreshold:
+		level = zerolog.WarnLevel
+	}
+
+	query = normalizeQueryString(query)
+	logging.L.WithLevel(level).
+		CallerSkipFrame(2). // logQuery + tx.{Query,Exec}
+		Int64("rows", rows).
+		Str("query", query).
```

Comment:
One thing we need to be mindful of when logging SQL queries is to omit query parameters lest they leak secrets or other information

Reply:
Right! This is the first argument to `Tx.{Query, QueryRow, Exec}` which can never include those values (because it would be a potential SQL injection attack if they did).